companyweb_base/models/companyweb_partner_model.py

```diff
@@ -0,0 +1,183 @@
+    cweb_balance_year_enable = fields.Boolean()
+    cweb_equityCapital = fields.Float("CompanyWeb Equity Capital", readonly=True)
+    cweb_addedValue = fields.Float("CompanyWeb Gross Margin (+/-)", readonly=True)
+    cweb_turnover = fields.Float("CompanyWeb Turnover", readonly=True)
+    cweb_result = fields.Float(
+        "CompanyWeb Fiscal Year Profit/Loss (+/-)", readonly=True
+    )
+    cweb_prefLang = fields.Char("CompanyWeb Preferred Language", readonly=True)
+    cweb_prefLang_enable = fields.Boolean()
+
+    def create_hash(self, login, password, secret):
+        today = datetime.now().strftime("%Y%m%d")
+        text = (today + login + password + secret).lower()
+        return sha1(text.encode("utf-8")).hexdigest()
+
+    def button_companyweb(self):
+        SERVICE_INTEGRATOR_ID = "acsone"
+        SERVICE_INTEGRATOR_SECRET = "ECAB8ACF-9AE1-4E90-BD0D-05A1F47A3FE9"
+        LOGIN = self.env.user.cweb_login
+        PASSWORD = self.env.user.cweb_password
+        _logger.debug("Current Partner TVA :  %s", self.vat)
+        wizard_form = self.env.ref(
+            "companyweb_base.companyweb_credential_wizard", False
+        )
+        _logger.debug(
+            "Id for the form : companyweb_base.companyweb_credential_wizard :  %s",
+            wizard_form.id,
+        )
+        if not LOGIN or not PASSWORD:
+            return {
+                "name": "No Credentials",
+                "type": "ir.actions.act_window",
+                "view_type": "form",
+                "view_mode": "form",
+                "res_model": "companyweb_base.credential_wizard",
+                "view_id": wizard_form.id,
+                "target": "new",
+            }
+
+        client = zeep.Client("https://connect.companyweb.be/V1.3/alacarteservice.asmx")
+        r = client.service.GetCompanyByVat(
+            dict(
+                CompanyWebLogin=LOGIN,
+                CompanyWebPassword=PASSWORD,
+                ServiceIntegrator=SERVICE_INTEGRATOR_ID,
+                LoginHash=self.create_hash(LOGIN, PASSWORD, SERVICE_INTEGRATOR_SECRET),
+                Language="FR",
+                VatNumber=self.vat,
+            )
+        )
+        _logger.debug("Return status for the cweb API call :  %s", r["StatusCode"])
+        wizard_form = self.env.ref(
+            "companyweb_base.companyweb_credential_wizard", False
+        )
+        if r["StatusCode"] == 101:
+            return {
+                "name": "Bad Credentials",
+                "type": "ir.actions.act_window",
+                "view_type": "form",
+                "view_mode": "form",
+                "res_model": "companyweb_base.credential_wizard",
+                "view_id": wizard_form.id,
+                "target": "new",
+            }
+        elif r["StatusCode"] == 306:
+            raise ValidationError(
+                self.vat + " is not a valid belgian vatnumber or empty."
+            )
```

Comment:
Should there be some `else` here ? Is there a status code for success ? What about other errors than 101 and 306 ?